src/WingetCreateTests/WingetCreateTests/UnitTests/UpdateCommandTests.cs

```diff
@@ -1,79 +1,118 @@
-        {
-            return new UpdateCommand
-            {
-                Id = id,
-                Version = version,
-                OutputDir = outputDir,
-                GitHubToken = this.GitHubApiKey,
-                WingetRepoOwner = this.WingetPkgsTestRepoOwner,
-                WingetRepo = this.WingetPkgsTestRepo,
-            };
-        }
-    }
-}
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateUnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Microsoft.WingetCreateCLI.Commands;
+    using Microsoft.WingetCreateCLI.Logging;
+    using Microsoft.WingetCreateCLI.Properties;
+    using Microsoft.WingetCreateCLI.Telemetry.Events;
+    using Microsoft.WingetCreateCore;
+    using Microsoft.WingetCreateCore.Common;
+    using Microsoft.WingetCreateCore.Models.Singleton;
+    using Microsoft.WingetCreateTests;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Test cases for verifying that the "update" command is working as expected.
+    /// </summary>
+    public class UpdateCommandTests : GitHubTestsBase
+    {
+        private const string UpdateCommandTestHeader = "WingetCreate Update Command Tests";
+        private readonly string tempPath = Path.GetTempPath();
+        private StringWriter sw;
+        private CommandExecutedEvent testCommandEvent;
+
+        /// <summary>
+        /// Setup method for the update command unit tests.
+        /// </summary>
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            this.testCommandEvent = new CommandExecutedEvent();
+            this.sw = new StringWriter();
+            Console.SetOut(this.sw);
+            Logger.Initialize();
+        }
+
+        /// <summary>
+        /// Teardown method for the update command unit tests.
+        /// </summary>
+        [OneTimeTearDown]
+        public void TearDown()
+        {
+            this.sw.Dispose();
+        }
+
+        /// <summary>
+        /// Verifies that the update command modifies the manifest's version field
+        /// when a "Version" argument is provided.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Test]
+        public async Task UpdateVersion()
+        {
+            string version = "1.2.3.4";
+            UpdateCommand command = this.GetUpdateCommand(TestConstants.TestPackageIdentifier, version, this.tempPath);
+            SingletonManifest testManifest = Serialization.DeserializeFromPath<SingletonManifest>(TestUtils.GetTestFile($"{TestConstants.TestPackageIdentifier}.yaml"));
+            List<string> manifests = new List<string> { testManifest.ToYaml(UpdateCommandTestHeader) };
+            manifests.Add(testManifest.ToYaml(UpdateCommandTestHeader));

```

Comment:
Unless you're wanting to explicitly test the yaml round-tripping, you could just change this to a new List<string> { File.Read(...yaml) }

Reply:
I think I agree that testing the yaml round-tripping is a bit excessive here. Changed to suggestion